internal/current_target.go

```diff
@@ -0,0 +1,58 @@
+package internal
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type CurrentTarget struct {
+	Name    string
+	Version int
+}
+
+func LoadCurrentTarget(currentTargeFile string) (CurrentTarget, error) {
+	cur := CurrentTarget{}
```

Comment:
Nitpick: I generally prefer `var cur CurrentTarget` or named return values like `func LoadCurrentTarget(currentTargeFile string) (cur CurrentTarget, err error)` depending onto which one saves more space.
IIUC these should work faster as more compiler optimizations are applicable to them.